### PR TITLE
feat(components/layout): box test harness

### PR DESCRIPTION
**Pull request by Blackbaud-SandhyaRajasabeson**
[#AB2195370](https://blackbaud.visualstudio.com/Products/_boards/board/t/SKY%20UX%20Program/Stories/?workitem=2195370)
[#AB2271298](https://blackbaud.visualstudio.com/Products/_boards/board/t/SKY%20UX%20Program/Stories/?workitem=2271298)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/1300/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/a11y-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/ag-grid-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/core-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/indicators-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/lists-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/modals-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/pages-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/popovers-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/tabs-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/1300/storybooks/tiles-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/1300/code-examples/)
- [dep-graph](https://blackbaud.github.io/skyux-pr-preview/1300/dep-graph/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/1300/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/1300/playground/)

File: apps/code-examples/src/app/code-examples/layout/box/basic/box-demo.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { SkyBoxHarness } from '@skyux/layout/testing';
+
+import { BoxDemoComponent } from './box-demo.component';
+import { BoxDemoModule } from './box-demo.module';
+
+fdescribe('Basic box', async () => {
```
Comment (Blackbaud-ErikaMcVey) [inline apps/code-examples/src/app/code-examples/layout/box/basic/box-demo.component.spec.ts]: left the `fdescribe` in here

File: libs/components/layout/testing/src/box/box-harness.filters.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { SkyHarnessFilters } from '@skyux/core/testing';
+
+/**
+ * A set of criteria that can be used to filter a list of SkyBoxHarness instances.
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/layout/testing/src/box/box-harness.filters.ts]: looks like we're not consistent about this but I think SkyBoxHarness should have back ticks around it so it appears formatted in the docs

File: libs/components/layout/testing/src/box/box-harness.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-box';
+
+  #getBox = this.locatorFor('.sky-box');
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyModalHarness` that meets certain criteria
```
Comment (Blackbaud-ErikaMcVey) [inline libs/components/layout/testing/src/box/box-harness.ts]: should be `SkyBoxHarness`, not modal

**Review by Blackbaud-ErikaMcVey — APPROVED** [review]

**Comment by blackbaud-sky-build-user** [comment]
Cherry-pick successful! :cherries: :tada: See https://github.com/blackbaud/skyux/pull/1303

**Pull request merged**